src/example-types/facet.js

```diff
@@ -134,5 +135,77 @@ module.exports = {
           }),
         options
       ),
-    })),
+    }))
+    // results.options.name is ObjectId which need to stringify to get the correct missedValues
+    // stringify values to avoid the bug when values are numeric values
+    let missedValues = _.difference(
+      _.map(_.toString, values),
+      _.map(({ name }) => _.toString(name), results.options)
+    )
+
+    let getSelectValues = node =>
+      node.isMongoId ? _.map(ObjectID, values) : values
+
+    let matchSelectedValues = node => ({
+      $match: { _id: { $in: getSelectValues(node) } },
+    })
+
+    if (!_.isEmpty(missedValues)) {
+      //create a values hash map for filter noValuesOptions
+      let getValuesIndex = _.reduce(
+        (acc, value) => {
+          acc[value] = true
+          return acc
+        },
+        {},
+        values
+      )
+
+      let noValuesOptions = _.reject(
+        record => getValuesIndex[record.name],
+        results.options
+      )
+      let dropSize = noValuesOptions.length + values.length - node.size
+      if (dropSize > 0) noValuesOptions = _.drop(dropSize, noValuesOptions)
+
+      let searchValues = await search(
+        _.compact([
+          matchSelectedValues(node),
```

Comment:
let's follow the same algorithm as contexture-elasticsearch - you should restrict this match to only the missing values (since we already have counts for many of them, it's more performant to just get the missing ones)

Reply:
I think we have bugs on contexture-elasticsearch.  Since we directly contact the missing value. the size of the result option will not match the original size.  for example if original the node.size in request is 20 and if we have 2 missing value. we will have 22 items in result options. Should I use the same method as contexture-elasticsearch?

Reply:
just documenting our conversation here - returning more options than `node.size` is totally fine. It's the right behavior as well since having a bunch of checked values would prevent you from getting more options (which is particularly wrong when you have an optionsFilter)